File: app/src/main/java/com/pashkobohdan/ttsreader/model/database/room/dao/BookDAO.java
package com.pashkobohdan.ttsreader.model.database.room.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.pashkobohdan.ttsreader.model.dto.book.BookDTO;
import com.pashkobohdan.ttsreader.model.dto.book.BookTitleImageDTO;

import java.util.List;

/**
 * Created by bohdan on 10.12.17.
 */

@Dao
public interface BookDAO{

    @Query("SELECT * FROM Book")
    List<BookDTO> getAllBookDtoList();

    @Query("SELECT * FROM BookTitleImage")
    List<BookTitleImageDTO> getAllBookTitleImageDtoList();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllBookDTO(BookDTO... bookDTOS);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllBookTitleImageDTO(BookTitleImageDTO... bookTitleImageDTOS);
}

File: app/src/main/java/com/pashkobohdan/ttsreader/model/database/room/dao/BookDAO.kt
package com.pashkobohdan.ttsreader.model.database.room.dao

import android.arch.persistence.room.*
import com.pashkobohdan.ttsreader.model.dto.book.BookDTO
import com.pashkobohdan.ttsreader.model.dto.book.BookTitleImageDTO

@Dao
interface BookDAO {

    @get:Query("SELECT * FROM Book")
    val allBookDtoList: List<BookDTO>

    @get:Query("SELECT * FROM BookTitleImage")
    val allBookTitleImageDtoList: List<BookTitleImageDTO>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllBookDTO(vararg bookDTOS: BookDTO)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllBookTitleImageDTO(vararg bookTitleImageDTOS: BookTitleImageDTO)

    @Delete
    fun deleteBook(vararg bookDTO: BookDTO)
}
